controllers/toolchainclustercache/healthchecker_test.go

```diff
@@ -1,177 +0,0 @@
-package toolchainclustercache
```

Comment:
it would be nice to preserve these tests and test the `HealthChecker` logic in isolation - i.e., outside its usage in the `Reconciler`.

I'd propose to retain these tests and refactor them to use the `updateIndividualClusterStatus` instead  `updateClusterStatuses`. Something like the following:

From:
```golang
	t.Run("ToolchainCluster.status doesn't contain any conditions", func(t *testing.T) {
		unstable, sec := newToolchainCluster("unstable", tcNs, "http://unstable.com", toolchainv1alpha1.ToolchainClusterStatus{})
		notFound, _ := newToolchainCluster("not-found", tcNs, "http://not-found.com", toolchainv1alpha1.ToolchainClusterStatus{})
		stable, _ := newToolchainCluster("stable", tcNs, "http://cluster.com", toolchainv1alpha1.ToolchainClusterStatus{})

		cl := test.NewFakeClient(t, unstable, notFound, stable, sec)
		reset := setupCachedClusters(t, cl, unstable, notFound, stable)
		defer reset()

		// when
		updateClusterStatuses(context.TODO(), "test-namespace", cl)

		// then
		assertClusterStatus(t, cl, "unstable", notOffline(), unhealthy())
		assertClusterStatus(t, cl, "not-found", offline())
		assertClusterStatus(t, cl, "stable", healthy())
	})
```

To:
```golang
	t.Run("ToolchainCluster.status doesn't contain any conditions", func(t *testing.T) {
               t.Run("unstable", func(t *testing.T) {
			unstable, sec := newToolchainCluster("unstable", tcNs, "http://unstable.com", toolchainv1alpha1.ToolchainClusterStatus{})

			cl := test.NewFakeClient(t, unstable, sec)
			reset := setupCachedClusters(t, cl, unstable)
			defer reset()

			// when
			err := updateIndividualClusterStatus(context.TODO(), unstable)

			// then
                        require.NoErr(t, err)
			assertClusterStatus(t, cl, "unstable", notOffline(), unhealthy())
                })

               t.Run("notFound", func(t *testing.T) {
			notFound, _ := newToolchainCluster("not-found", tcNs, "http://not-found.com", toolchainv1alpha1.ToolchainClusterStatus{})

			cl := test.NewFakeClient(t, notFound, sec)
			reset := setupCachedClusters(t, cl, notFound)
			defer reset()

			// when
			err := updateIndividualClusterStatus(context.TODO(), notFound)

			// then
                        require.NoErr(t, err)
			assertClusterStatus(t, cl, "not-found", offline())
                })

               t.Run("stable", func(t *testing.T) {
			stable, _ := newToolchainCluster("stable", tcNs, "http://cluster.com", toolchainv1alpha1.ToolchainClusterStatus{})

			cl := test.NewFakeClient(t, stable, sec)
			reset := setupCachedClusters(t, cl, stable)
			defer reset()

			// when
			err := updateIndividualClusterStatus(context.TODO(), stable)

			// then
                        require.NoErr(t, err)
			assertClusterStatus(t, cl, "stable", healthy())
                })
	})
```

Reply:
I have already done that, but i used  reconciler instead of `updateIndividualClusterStatus` https://github.com/codeready-toolchain/toolchain-common/pull/386/files#diff-0ba9f4ceb68c0237cce25b4e5cec790d3d8344311cf62fc0a614411eb2fa956aR174 
But I think i can add one more case to test `updateIndividualClusterStatus`

Reply:
updated it